dev/com.ibm.ws.security.mp.jwt/src/com/ibm/ws/security/mp/jwt/impl/utils/ClaimsUtils.java

```diff
@@ -40,29 +42,50 @@
 
     private static final JsonBuilderFactory builderFactory = Json.createBuilderFactory(null);
 
-    public ClaimsUtils() {
-    }
-
     /**
      * Parses the provided JWT and returns the claims found within its payload.
      */
-    public JwtClaims getJwtClaims(String jwt) throws JoseException {
+    public static JwtClaims getJwtClaims(String jwt) throws JoseException {
         String methodName = "getJwtClaims";
         if (tc.isDebugEnabled()) {
             Tr.entry(tc, methodName, jwt);
         }
-        JwtClaims jwtclaims = new JwtClaims();
+        JwtClaims jwtclaims;
         String payload = getJwtPayload(jwt);
         if (payload != null) {
             jwtclaims = getClaimsFromJwtPayload(jwt, payload);
+        } else {
+            jwtclaims = new JwtClaims();
+        }
+        if (tc.isDebugEnabled()) {
+            Tr.exit(tc, methodName, jwtclaims);
         }
+        return jwtclaims;
+    }
+
+    public static JwtClaims getJwtClaims(JwtToken jwtToken) {
+        String methodName = "getJwtClaims";
+        if (tc.isDebugEnabled()) {
+            Tr.entry(tc, methodName, jwtToken);
+        }
+        JwtClaims jwtclaims = new JwtClaims();
+        Claims claims = jwtToken.getClaims();
+        Set<Entry<String, Object>> entries = claims.entrySet();
+        for (Entry<String, Object> entry : entries) {
+            addEntryToClaims(entry, jwtclaims);
+        }
+        jwtclaims.setStringClaim(org.eclipse.microprofile.jwt.Claims.raw_token.name(), jwtToken.compact());
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "Key : " + "raw_token" + ", Value: " + "raw_token");
+        }
+        convertJoseTypes(jwtclaims);
```

Comment:
This might be better extracted to its own method. When doing the code review I wasn't quite sure why this was here. It wasn't until I looked at the code as it is now that I realized that this is part of what `getClaimsFromJwtPayload()` does, and therefore an implicit part of the `getJwtClaims(String)` method. These lines have to be here to ensure parity between the two `getJwtClaims` methods. Moving these lines to their own method would make it clearer that both methods perform these steps.

Reply:
I had thought similarly.  I have updated the PR with some refactoring.  It is a bit more than what you are suggesting.